mriqc/viz/fmriplots.py

```diff
@@ -135,24 +135,34 @@ def fmricarpetplot(func_data, segmentation, outer_gs, tr=None, nskip=0):
 
     # Carpet plot
     ax1 = plt.subplot(gs[1])
-    ax1.imshow(detrended[order, :], interpolation='nearest',
+
+    # Avoid segmentation faults for long acquisitions by decimating the input data
+    long_cutoff = 800
+    if detrended.shape[1] > long_cutoff:
+        data = detrended[order, ::2]
+    else:
+        data = detrended[order, :]
+
+    ax1.imshow(data, interpolation='nearest',
```

Comment:
I use it here scaling: https://github.com/poldracklab/mriqc/blob/master/mriqc/viz/utils.py#L58

But yeah, lets go away with this, and I'll open an issue to revise this in case we don't have anything else with higher priority